src/Apps/Sell/Routes/SubmissionRoute.tsx

```diff
@@ -25,7 +26,10 @@ export const SubmissionRoute: React.FC<SubmissionRouteProps> = props => {
   return (
     <FullBleed>
       <AppContainer>
-        <SellFlowContextProvider submissionID={submission.externalId}>
+        <SellFlowContextProvider
+          submissionID={submission.externalId}
+          submissionState={submission.state}
```

Comment:
We are saving the submission state in the context to prevent updating an already submitted submission because this would result in an error.

Reply:
Instead of persisting the initial state here, can we better respond to the mutation error? What kind of message do we get back from the backend? Would that be revealing enough to expose an error via the same toast mechanism?

That said, we should have a broader discussion on possible submission states and which states allow updates to take place. For example, we're planning to allow updates after a submission is `approved` but are we considering it "locked" between `submitted` and `approved`? Might be useful to discuss and align on expectations.

Reply:
When the submission is in the state "SUBMITTED", updating the submission returns the error "Submission Not Found". So it's possible to load the submission but not to update it. 

What do you think about merging this as an interim solution until we have decided whether we allow updating submitted/approved submissions? This isn't ideal, but at least it returns a reasonable error message for now. Just in case a user somehow lands directly on a step for an already submitted submission.

Alternatively, we could render an error message in this case and not even allow viewing the step 🤔

Reply:
> Submission Not Found

Hmm! Sounds like some system (probably Convection) is trying to find the submission in a draft state right away and throws a 404 if not found.

I would expect that if the submission ID is valid but just in the wrong state for the attempted action (updating), that we could return a `422: Unprocessable Entity` error instead along with a more revealing error message. So we could make that improvement to make the client implementation a bit simpler.

I suppose if we don't support any editing on the submission in certain states — is there another surface we could redirect to? Maybe should the error message there? Thinking of the My Collection artwork page or even the My Collection landing surface...

Reply:
You're right, the `updateConsignmentSubmission` resolver in Convection returns the incorrect error message ([GitHub](https://github.com/artsy/convection/blob/45f7d5984bc26ad90a9ed8ef92bb28f60facccfd/app/graphql/resolvers/update_submission_resolver.rb#L9)). Updating the error message in Convection would allow us to simplify this PR so that we don't have to check the submission's state in the client.

I'd say we go this route: correct the error message in Convection and display it if updating the submission fails.

Reply:
https://github.com/artsy/convection/pull/1483